### PR TITLE
[ Rel-5_0 Bug 11819 ] - Error when trying to edit/create new customer agent while SMIME activated 

### PR DESCRIPTION
**Pull request by s7design**
http://bugs.otrs.org/show_bug.cgi?id=11819

Hi Martin,

There was a problem because there did not updated some modules in the script  DBUpdate-to-5.pl
Now, I check all sysconfig items which have Module param. I hope that now it works for all items. 

If you want I can split this PR in two ones, because there is fixed bug with changes for   "Preferences Modules" and fixed "Article View Modules" that is not mentioned in the bug. I thought that it is better solve in one PR. 

Regards
Zoran

### CODE REVIEW

File: scripts/DBUpdate-to-5.pl [hunk 1]
```diff
@@ -1339,33 +1339,45 @@ sub _MigrateConfigs {
     print "...done.\n";
     print "--- Preferences group modules...";
 
-    # Preferences groups
-    $Setting = $ConfigObject->Get('PreferencesGroups');
+    # Preferences Modules
+    my @PreferencesGroupTypes = (
+        {
+            Path => 'PreferencesGroups',
```
Comment (mgruner) [inline scripts/DBUpdate-to-5.pl]: This looks very good in general. However, I have one question: why do you use a list of hashes here? Would not a simple list ```@PreferencesGroupTypes = qw(PreferencesGroups CustomerPreferencesGroups)``` be enough? I see that it is already used on other places in this function, but maybe you could consider a simplification.

Comment (s7design) [inline scripts/DBUpdate-to-5.pl]: Thanks Martin, you are right. I will change it and update this PR very soon.
Regards
Zoran

**Comment by s7design** [comment]
Hi @mgruner 

I updated this PR as you suggested

Regards
Zoran

**Comment by mgruner** [comment]
Very nice, thanks! I have one more small request: the settings should only be updated if they actually change (Module vs. NewModule).

**Comment by s7design** [comment]
Hi @mgruner,

Thanks for your suggestion, I think it is better now to update only sysconfig which are changed. Please let me know if my proposal good now.

Regards
Zoran

**Comment by mgruner** [comment]
Merged with slight changes, I modified two more places (cluster configuration and customer queue selection generic) to only update if needed. Please let me know in case you see any problem there. Thanks!

**Comment by s7design** [comment]
Thanks Martin, I tested it and it works fine. I agree with you it is better now with your changes.

Regards
Zoran

**Pull request closed without merging**